src/ast/nodes/ClassDeclaration.ts

```diff
@@ -40,7 +40,7 @@ export default class ClassDeclaration extends ClassNode {
 				code.appendLeft(this.end, `${_}${getSystemExportStatement([variable], options)};`);
 			}
 			const renderedVariable = variable.getName(getPropertyAccess);
-			if (renderedVariable !== name) {
+			if (renderedVariable !== name && this.decorators.length === 0) {
```

Comment:
While I cannot prove it, I think the code that is generated without this change

```js
let Foo$1 = @fooDecorator
class Foo {};
```

might look weird but is actually correct. And it would preserve the class name. So maybe we revert this change?

Reply:
Yeah 😄, I tried to run the compiled code generated by TS for the above code, and the result is correct.
https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABAZwKYEMA2qAmAKASkQG8AoRRCBZObAOkzgHM8BGAZlYCZOCBucogBOqKCCFI8VMMihCQ0OEIBciAGLho8MEQC8APhKCK0mvUYsuVq-2OIA8gCMAVqmh00WKdTkKoS2woKJ1d3T0xvGV9FIToAByE4fygATzjUQMQAX1Ic7Ch1ODgAElZEXUQAAXDcQlIITHRkZEK4Emy+IA

Reply:
But the right part of `let Foo$1 = @fooDecorator` should be the deconflicted name.

Reply:
Oh, The previous code didn't take this point into account either. I will fix it soon.